tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+                                o.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut o = o.up(helper)?;
+                            o.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Operator(o));
+                        }
+                        StmtRaw::Script(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut s = s.up(helper)?;
+                            s.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Script(s));
+                        }
+                        StmtRaw::Stream(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            query_stmts.push(Stmt::Stream(s.up(helper)?));
+                        }
+                        StmtRaw::Select(mut s) => {
+                            // TTODO Inline args in where, having, groupby clauses
+                            s.from.0.id = Cow::owned(self.mangle_id(&s.from.0.id.to_string()));
+                            s.into.0.id = Cow::owned(self.mangle_id(&s.into.0.id.to_string()));
+                            // TTODO This leads to ugly error messages if stream is not defined.
+                            // `Stream used in `from` or `into` is not defined: A__SUBQ__custom_subquery_out`
+                            query_stmts.push(StmtRaw::Select(s).up(&mut helper)?);
+                        }
+                        StmtRaw::ScriptDecl(mut s) => {
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            query_stmts.push(StmtRaw::ScriptDecl(s).up(&mut helper)?);
+                        }
+                        StmtRaw::OperatorDecl(mut o) => {
+                            o.params = SubqueryStmtRaw::inline_maybe_params(
+                                o.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            query_stmts.push(StmtRaw::OperatorDecl(o).up(&mut helper)?);
+                        }
+                        StmtRaw::SubqueryDecl(mut s) => {
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
```

Comment:
A short documentation would be nice on how args in nested subqueries behave, I believe they're not recursively resolved right?